lib/membrane_mp4/muxer/cmaf.ex

```diff
@@ -403,14 +402,10 @@ defmodule Membrane.MP4.Muxer.CMAF do
       |> Map.values()
       |> Enum.all?(& &1.end_of_stream?)
 
-    if SamplesQueue.empty?(cache) do
-      if processing_finished? do
-        end_of_streams = generate_output_end_of_streams(ctx)
+    if SamplesQueue.empty?(cache) and processing_finished? do
+      end_of_streams = generate_output_end_of_streams(ctx)
 
-        {end_of_streams, state}
-      else
-        {[redemand: output_pad], state}
-      end
+      {end_of_streams, state}
```

Comment:
- this still doesn't guarantee we don't demand on an EOSed pad
- are you sure we never need this redemand here?

Are you sure we don't need to do a redemand

Reply:
I didn't change this particular clause at all just to be clear, and yes we don't need to do redemand in this case because there is no input pad left and the `SampleQueue` is empty.

Reply:
When `processing_finished?` is false, some input pads are still left. Previously we redemanded when `processing_finished?` was `false`. Now we don't unless I'm missing something.

> this still doesn't guarantee we don't demand on an EOSed pad why is that?

Redemanding means that handle_demand is executed immediately if there's still some demand on output. But if we don't redemand, handle_demand can still be executed right away - it's up to the framework. In that case, your change doesn't make any difference from the current behaviour.

Reply:
when it comes to demanding withoug `redemand` the only situation in which we don't send redemand is when we send end of stream on output